app/Policies/UserPolicy.php

```diff
@@ -3,63 +3,32 @@
 namespace App\Policies;
 
 use App\Models\User;
-use Illuminate\Auth\Access\HandlesAuthorization;
 
-class UserPolicy
+class UserPolicy extends BasePolicy
 {
-	use HandlesAuthorization;
-
-	public const IS_ADMIN = 'isAdmin';
-	public const CAN_UPLOAD = 'canUpload';
-	public const CAN_EDIT_SETTINGS = 'canEditSettings';
-
-	/**
-	 * Perform pre-authorization checks.
-	 *
-	 * @param \App\Models\User $user
-	 * @param string           $ability
-	 *
-	 * @return void|bool
-	 */
-	public function before(?User $user, $ability)
-	{
-		if ($this->isAdmin($user)) {
-			return true;
-		}
-	}
+	public const CAN_EDIT_OWN_SETTINGS = 'canEditOwnSettings';
+	public const CAN_CREATE_OR_EDIT_OR_DELETE = 'canCreateOrEditOrDelete';
+	public const CAN_LIST = 'canList';
 
 	/**
-	 * This defines if the user is admin.
+	 * This defines if user can edit their settings.
 	 *
-	 * @param User|null $user
+	 * @param User $user
 	 *
 	 * @return bool
 	 */
-	public function isAdmin(?User $user): bool
+	public function canEditOwnSettings(User $user): bool
 	{
-		return $user?->id === 0;
+		return $user->may_edit_own_settings;
 	}
 
-	/**
-	 * This defines if user can edit their settings.
-	 *
-	 * @param User $user
-	 *
-	 * @return bool
-	 */
-	public function canEditSettings(User $user): bool
+	public function canCreateOrEditOrDelete(User $user): bool
 	{
-		return !$user->is_locked;
+		// Note, the administrator is already handled in the `before()` method and every one else is not allowed to create/delete users.
+		return false;
 	}
```

Comment:
I still don't understand why we need the extra _right_ `canEditOwnSettings`. As a capability? Yes, sure. As a right? No.

IMHO, the code should look like this

```php
public function canEdit(User $user, ?User $user2): bool
{
	// Note, the administrator is already handled in the `before()` method, hence we only need to deal with the case of self-editing here
	return $user->may_edit_own_settings && $user->id === $user2?->id;
}

public function canCreateOrDelete(User $user): bool
{
	// Note, the administrator is already handled in the `before()` method and every one else is not allowed to create/delete users.
	return false;
}
```

Reply:
Refactored.